### ISSUE TITLE
Auto-Unseal with Azure Key Vault doesn't work

### ISSUE THREAD

**Issue by mxplusb** [hashicorp/vault-helm#54]
This is the same behaviour as hashicorp/vault#6959, just from the Helm chart. I have pre-deployed the Consul Helm chart and am using that as the backend config.

With this `values.yml`:

```yaml
global:
  enabled: true
  image: "vault:1.2.3"

server:
  resource: |
    requests:
      memory: "512Mi"
    limits:
      memory: "1Gi"
  enabled: true

  service:
    enabled: true

  ha:
    enabled: true
    replicas: 3

    config: |
      ui = true

      listener "tcp" {
        tls_disable = 1
        address = "[::]:8200"
        cluster_address = "[::]:8201"
      }

      storage "consul" {
        path = "vault"
        address = "ap-discovery-consul-server-0.ap-discovery-consul-server.infrastructure.svc.cluster.local:8500"
      }

      seal "azurekeyvault" {
        client_id      = "<client-id>"
        client_secret  = "<client-secret>"
        tenant_id      = "<tenant-id>"
        vault_name     = "<keyvault-name>"
        key_name       = "<unseal-key>"
      }
```

With `v0.1.2` of the Helm chart:

```bash
$ helm install --atomic --name ap-secrets -f <...>\Vault.yaml --namespace infrastructure .

# in other terminal
% kubectl logs -f ap-secrets-vault-0 --namespace infrastructure -c vault                                           ==> Vault server configuration:

       Azure Environment: AzurePublicCloud
          Azure Key Name: <unseal-key>
        Azure Vault Name: <keyvault-name>
               Seal Type: azurekeyvault
             Api Address: http://10.200.33.50:8200
                     Cgo: disabled
         Cluster Address: https://10.200.33.50:8201
              Listener 1: tcp (addr: "[::]:8200", cluster address: "[::]:8201", max_request_duration: "1m30s", max_request_size: "33554432", tls: "disabled")
               Log Level: info
                   Mlock: supported: true, enabled: true
                 Storage: consul (HA available)
                 Version: Vault v1.2.3

2019-09-14T06:01:20.848Z [WARN]  storage.consul: appending trailing forward slash to path
==> Vault server started! Log data will stream in below:

2019-09-14T06:01:22.177Z [INFO]  core: stored unseal keys supported, attempting fetch
2019-09-14T06:01:22.192Z [WARN]  failed to unseal core: error="stored unseal keys are supported, but none were found"
2019-09-14T06:01:23.691Z [INFO]  core: autoseal: seal configuration missing, but cannot check old path as core is sealed: seal_type=recovery
2019-09-14T06:01:26.685Z [INFO]  core: autoseal: seal configuration missing, but cannot check old path as core is sealed: seal_type=recovery
2019-09-14T06:01:27.192Z [INFO]  core: stored unseal keys supported, attempting fetch
2019-09-14T06:01:27.198Z [WARN]  failed to unseal core: error="stored unseal keys are supported, but none were found"
2019-09-14T06:01:29.711Z [INFO]  core: autoseal: seal configuration missing, but cannot check old path as core is sealed: seal_type=recovery
...
```

Attempting to unseal it:

```bash
% kubectl exec -ti pods/ap-secrets-vault-0 --namespace infrastructure sh                                           Defaulting container name to vault.                                                                                               Use 'kubectl describe pod/ap-secrets-vault-0 -n infrastructure' to see all of the containers in this pod.
/ # vault status                                                                                                                  Key                      Value
---                      -----
Recovery Seal Type       azurekeyvault
Initialized              false
Sealed                   true
Total Recovery Shares    0
Threshold                0
Unseal Progress          0/0
Unseal Nonce             n/a
Version                  n/a
HA Enabled               true
/ # vault operator init
Error initializing: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/sys/init
Code: 400. Errors:

* Vault is already initialized
```

I did double check my unseal key policies in Azure Key Vault, and it has all permissions minus `delete`, so I know it's not a permission's issue.

**Comment by shotah** [comment]
tl;dr Are you swapping out the variables in that output or is it outputting that way? Just a quick glance would suggest your variables aren't getting replaced.

**Comment by mxplusb** [comment]
I'm swapping out variables in the pasted text for security reasons (sorry). They match the input variables, though.

**Comment by mxplusb** [comment]
Filed hashicorp/vault#7481 as it looks to be an issue with Vault, not the chart as this can be reproduced locally.

**Comment by jasonodonnell** [comment]
Closing since unrelated to Vault Helm, thanks for the report!
